src/kv/store.h

```diff
@@ -377,31 +410,35 @@ namespace kv
         {
           return success;
         }
+
         auto h = get_history();
-        if (h)
+
+        auto search = views.find("ccf.signatures");
+        if (search != views.end())
         {
-          auto search = views.find("ccf.signatures");
-          if (search != views.end())
+          // Transactions containing a signature must only contain
+          // a signature and must be verified
+          if (views.size() > 1)
           {
-            // Transactions containing a signature must only contain
-            // a signature and must be verified
-            if (views.size() > 1)
-            {
-              LOG_FAIL_FMT("Failed to deserialize");
-              LOG_DEBUG_FMT(
-                "Unexpected contents in signature transaction {}", v);
-              return DeserialiseSuccess::FAILED;
-            }
+            LOG_FAIL_FMT("Failed to deserialize");
+            LOG_DEBUG_FMT("Unexpected contents in signature transaction {}", v);
+            return DeserialiseSuccess::FAILED;
+          }
 
+          if (h)
+          {
             if (!h->verify(term))
             {
               LOG_FAIL_FMT("Failed to deserialize");
               LOG_DEBUG_FMT("Signature in transaction {} failed to verify", v);
               return DeserialiseSuccess::FAILED;
             }
-            success = DeserialiseSuccess::PASS_SIGNATURE;
           }
+          success = DeserialiseSuccess::PASS_SIGNATURE;
```

Comment:
To clarify the changes here - it is now possible to return `PASS_SIGNATURE` when there is no `history`. If there _is_ a history we will use it to verify the signature, but in unit tests where there isn't we can accept stub signatures.